runway/blueprints/staticsite/staticsite.py

```diff
@@ -491,92 +463,68 @@ def add_cloudfront_directory_index_rewrite_version(self, directory_index_rewrite
             dict: The CloudFront directory index rewrite version
 
         """
-        # Generating a unique resource name here for the Lambda version, so it
-        # updates automatically if the lambda code changes
-        code_hash = hashlib.md5(
-            str(directory_index_rewrite.properties['Code'].properties['ZipFile'].to_dict()).encode()  # noqa pylint: disable=line-too-long
-        ).hexdigest()
-
+        s3_key = directory_index_rewrite.properties['Code'].to_dict()['S3Key']
+        code_hash = s3_key.split('.')[0].split('-')[-1]
```

Comment:
I think my advice here ended up being incorrect -- since this function is being directly embedded into the template, there's no "S3Key" on the dict (it's just `{'ZipFile': 'foo'}`).

Reply:
Reverted